File: w06/d04/Andrew_Wallace/hangman.js
function Hangman(word) {
  var wrongGuess = false;
  var badGuessCount = 0;
  var gameOver = false;
  var playerProgress = []
  var wordLength = word.length;
  var letters = word.split('');

 for (var i = 0; i < word.length; i++) {
      playerProgress.push("_");
  }

  this.progress = function() {
    return playerProgress.join(" ");
  }

  this.guess = function(letter) {
    wrongGuess = true;
    var placeholders = playerProgress;
    for (var i = 0; i <= word.length; i++) {
      if (word.charAt(i) === letter.toLowerCase() && badGuessCount <= 5) {
        wrongGuess = false;
        placeholders[i] = letter;
      } else {
      badGuessCount++;
      }
    }
  }
}

  // this.guess = function(letter) {
  //   wrongGuess = true;
  //   var placeholders = playerProgress;
  //   for (var i = 0; i < word.length; i++) {
  //     if (word.charAt(i) == letter.toLowerCase() && badGuesses <= 5) {
  //       wrongGuess = false;
  //       placeholders[i] = letter;
  //     } else if ( badGuesses > 5) {
  //       return "Tough luck, you lose."
  //     } else {
  //       badGuesses++;
  //     }
  //   }
  // }











}

File: w10/d01/Andrew_Wallace/hangman/lib/hangman.rb
class Hangman

  def initialize(word)
    @word = word.split("")
    @progress = @word.map { "_" }
  end

  def progress()
    @progress.join(" ")
  end

  def guess(guessed_letter)
    if @word.include?(guessed_letter)
      @word.each_with_index do |letter, index|
        if letter == guessed_letter
          @progress[index] = letter
        end
      end
      return true
    else
      return false
    end
  end

  def lose_status
    @bad_guess_count = 6
    if @bad_guess_count > 5
      return "You lost!"
    else
      return "Keep guessing!"
    end
  end

  def win_status
    if @bad_guess_count <= 5 || progress
      return
    end
  end
end
